src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -1,28 +1,35 @@
 use crate::{
     buffers::Acker,
-    event::{Event, Metric},
+    event::{metric::Direction, Event, Metric},
     region::RegionOrEndpoint,
     sinks::util::{
         retries::{FixedRetryPolicy, RetryLogic},
-        BatchServiceSink, Partition, SinkExt,
+        BatchServiceSink, SinkExt,
     },
     topology::config::{DataType, SinkConfig},
 };
+use chrono::{DateTime, SecondsFormat, Utc};
 use futures::{Future, Poll};
 use rusoto_cloudwatch::{
     CloudWatch, CloudWatchClient, MetricDatum, PutMetricDataError, PutMetricDataInput,
 };
 use rusoto_core::{Region, RusotoFuture};
 use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
 use std::{convert::TryInto, time::Duration};
 use tower::{Service, ServiceBuilder};
-use tracing::field;
 use tracing_futures::{Instrument, Instrumented};
 
+#[derive(Clone, Default)]
+struct CloudWatchMetricsState {
```

Comment:
very minor nit but this name is getting a bit long, should we think about maybe just renaming it to `State` since its just an internal private struct?